packages/common/src/utils/filters.ts

```diff
@@ -421,6 +421,125 @@ export const getFilterRulesByFieldType = (
         },
     );
 
+const flattenFilterGroup = (filterGroup: FilterGroup): FilterGroup => {
+    const items = getItemsFromFilterGroup(filterGroup);
+
+    return {
+        id: filterGroup.id,
+        [getFilterGroupItemsPropertyName(filterGroup)]: items.reduce<
+            FilterGroupItem[]
+        >((acc, item) => {
+            if (isFilterGroup(item)) {
+                const flatGroup = flattenFilterGroup(item);
+
+                // If the current group is an AND group and the parent group is an AND group as well, we can flatten it's items
+                if (
+                    isAndFilterGroup(flatGroup) &&
+                    isAndFilterGroup(filterGroup)
+                ) {
+                    return [...acc, ...flatGroup.and];
+                }
+
+                // If the parent group type doesn't match the current group type or we're working with and OR group, we can't flatten it
+                return [...acc, flatGroup];
+            }
+
+            return [...acc, item];
+        }, []),
+    } as FilterGroup;
+};
+
+export const getFiltersFromGroup = (
+    filterGroup: FilterGroup,
+    fields: Field[],
+): Filters => {
+    const flatFilterGroup = flattenFilterGroup(filterGroup);
+    const items = getItemsFromFilterGroup(flatFilterGroup);
+    return items.reduce<Filters>((accumulator, item) => {
+        if (isFilterRule(item)) {
+            const fieldInRule = fields.find(
+                (field) => fieldId(field) === item.target.fieldId,
+            );
+
+            if (fieldInRule) {
```

Comment:
The code is fairly straight-forward, but some short comments explaining the business reasoning behind each of these branches would be great too 💅

Reply:
I've added the comments to both functions 👌 
Also fixed a slight bug regarding the filter group type https://github.com/lightdash/lightdash/pull/9743/commits/9530d022a018b6eab87410e4581ad153707282ce - see commit description